src/main/java/com/bmc/truesight/saas/meter/client/model/Event.java

```diff
@@ -34,20 +42,42 @@ public Instant timestamp() {
     public abstract Optional<String> at();
     public abstract Optional<String> ad();
     public abstract Optional<String> sender();
+    public abstract Optional<Map<String, String>> properties();
 
+    /**
+     * Creates an Event with the given required parameter: title.
+     *
+     * @param title the Title of this event (must not be greater than 255 characters long).
```

Comment:
Two Precondition checks are already in place using checkArgument in the protected void check() method.  I think those should be sufficient or did you want me to add an additional checkState as noted in the example URL?